runtime/builtin_types_test.go

```diff
@@ -422,3 +423,63 @@ func TestBuiltinSetAttr(t *testing.T) {
 		}
 	}
 }
+
+func TestRawInput(t *testing.T) {
+	fun := newBuiltinFunction("TestRawInput", func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+		// Create a fake Stdin for input test.
+		stdinFile, _ := ioutil.TempFile(os.TempDir(), "prefix")
```

Comment:
Not sure if it'd be easier, but you could also use a Pipe() instead of a temp file for stdin, similar to what you've done for stdout. Something like this:

```
stdinFile, w := os.Pipe()
go func() {
  w.Write([]byte("hello grumpy\n"))
  w.Close()
}()
```

You need to do the write asynchronously this way in case the Write() call blocks waiting for a Read() on the other end of the pipe.